### ISSUE TITLE
Input files to Image3Pipeline are deleted

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#8128]
_Issue [JP-3488](https://jira.stsci.edu/browse/JP-3488) was created on JIRA by [Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert):_

I'm running the Image3Pipeline with a set of NIRCam imaging mode files using a slightly non-standard (but comon) method. The default alignment in MAST for my data is poor, so I start with the cal.fits files and:

1. Run tweakreg manually with updated paramters. This produces a list of *tweakreg.fits files
2. Create an updated association file that contains the *tweakreg.fits files
3. Call Image3Pipeline, skipping tweakreg, and supplying the updated association file

The pipeline completes successfully, and the resulting i2d output looks good. However, either at the end of outlier_detection, or during the resample step, all of the input *tweakreg.fits files disappear from my directory. Prior to that point in the pipeline, the *tweakreg.fits files are still present. My guess is that the tweakreg files were renamed to, or deleted and replaced by, the *outlier_i2d.fits files that appear in the output directory? I'm guessing this could be a side effect of suffix replacement, since the input files contain a recognized suffix in the filenames?

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=676067#comment-676067):_

Nearly identical issue also reported in [INC0196153](https://stsci.service-now.com/nav_to.do?uri=%2Fincident.do%3Fsys_id%3D6623c85e97437d5423a2fe971153af14%26sysparm_stack%3Dincident_list.do%3Fsysparm_query%3Dactive%3Dtrue%26sysparm_view%3DJWST%26sysparm_view_forced%3Dtrue)

**Comment by stscijgbot-jp** [comment]
_Comment by [Armin Rest](https://jira.stsci.edu/secure/ViewProfile.jspa?name=arest) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=677720#comment-677720):_

I also have the same issue. This makes doing this very cumbersome....

**Comment by stscijgbot-jp** [comment]
_Comment by [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=682361#comment-682361):_

[Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert) can you please put your tweakreg.fits files somewhere for me to grab and do some testing? thank you

**Comment by stscijgbot-jp** [comment]
_Comment by [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=682599#comment-682599):_

[Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert) and/or [Armin Rest](https://jira.stsci.edu/secure/ViewProfile.jspa?name=arest) can you please try this branch and see if your files are deleted?
```java
pip install git+https://github.com/penaguerrero/jwst@twkregfix```

**Comment by stscijgbot-jp** [comment]
_Comment by [Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=683569#comment-683569):_

I don't have my files from last time, since outlier_detection deleted them. :) But I just created a few more tweakreg.fits files, and ran the rest of image3 with them, using your branch. The results look good to me. The tweakreg files are still present in the directory.

**Comment by stscijgbot-jp** [comment]
_Comment by [Maria Pena-Guerrero](https://jira.stsci.edu/secure/ViewProfile.jspa?name=pena) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=683694#comment-683694):_

thank you [Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert) !

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=683846#comment-683846):_

Fixed by #8263

**Comment by stscijgbot-jp** [comment]
_Comment by [Bryan Hilbert](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hilbert) on [JIRA](https://jira.stsci.edu/browse/JP-3488?focusedCommentId=684813#comment-684813):_

Tests confirm that the fix is working.
